src/vbd_store.ml

```diff
@@ -0,0 +1,37 @@
+open Lwt.Infix
+
+let m = Lwt_mutex.create ()
+
+let create_dir_if_doesnt_exist () =
+  Lwt.catch
+    (fun () -> Lwt_unix.mkdir Consts.xapi_nbd_persistent_dir 0o755)
+    (function
+      | Unix.(Unix_error (EEXIST, "mkdir", dir)) when dir = Consts.xapi_nbd_persistent_dir -> Lwt.return_unit
+      | e ->
+        Lwt_log.error_f "Failed to create directory: %s" (Printexc.to_string e)
+    )
+
+let transform_vbd_list f =
+  Lwt_mutex.with_lock m (fun () ->
+      create_dir_if_doesnt_exist () >>= fun () ->
+      (try
```

Comment:
Why `try` instead of `Lwt.catch`?

Reply:
Good spot. `Lwt_stream`s are a bit unique because they are not an `Lwt.t`, so that's why I originally used `try`, but now that I convert them into a list, we do get a `Lwt.t` type back, so we need `Lwt.catch` indeed.

Reply:
you mean that depending on the failure we need both try and catch? :O

Reply:
`Lwt.catch` is always enough, as it works with normal exceptions too as far as I know :)